pkg/infra/game/state/state.go

```diff
@@ -40,6 +40,23 @@ type AgentState struct {
 	Defector    Defector
 }
 
+func (s *AgentState) HasItem(itemType commons.ItemType, itemID commons.ItemID) bool {
+	var inventory immutable.List[Item]
+	if itemType == commons.Weapon {
```

Comment:
I think replacing this itemType check with a switch could make sense, or we change the `commons.ItemType` type alias to `bool` to avoid having default cases elsewhere.

Reply:
yeah, `bool` sounds good